source/ControlledActivated.tsx

```diff
@@ -122,7 +150,7 @@ const ControlledActivated = ({
   }, [isLoaded, onLoad, onUnload, prevIsLoaded])
 
   // use parent element or fake one if it's not yet loaded
-  const parentEl = parentRef.current || pseudoParentEl
+  const parentEl = parentRef?.current || pseudoParentEl
```

Comment:
I don't think this needs a `?`, for `useRef` from the parent will _always_ be `{ current }`

Reply:
@rpearce I believe refs are not always defined (in their life-cycle) so this check is necessary.

Reply:
@rpearce Never mind, TypeScript doesn’t complain in this use case.

Reply:
I think at somepoint when I was migrating, it complained, that's why I put it there. Maybe the type has made it unnecessary.

Reply:
Really? I've never had an issue on the frontend with that. Check this out:

> The returned object will persist for the full lifetime of the component.

https://reactjs.org/docs/hooks-reference.html#useref

I could be mistaken, but the value of `current` may or may not be there, but an object with a `current` property should (what `parentRef` is).

If I'm wrong on this, that would be good to know, too!